src/Map.cpp

```diff
@@ -150,46 +153,44 @@ bool Map::update() {
     	return false; // Game over 
     }
 
-    // check if ball hits player paddle, TODO add player1 to the Map, add list of players
+    // check if ball hits player paddle
     if (SDL_HasIntersection(player1_rect, ball_rect)) {
-        // TODO make function for changing ball velocity
 	velocity_y *= -1;
         velocity_x = rand() % Ball::ANGLE_RANGE - (Ball::ANGLE_RANGE-1) / 2;
         ball->updateVelocity(velocity_x, velocity_y);
     }
-     
+    
+    // check upper boundary
+    if (ball_y < Game::SCORE_HEIGHT) {
+    	ball->updatePosition(ball_x, Game::SCORE_HEIGHT + 5);
```

Comment:
just realizing this value of 5 looks so random - we should have that as a clearly defined class member variable